titus-server-master/src/main/java/com/netflix/titus/master/scheduler/TaskPlacementRecorder.java

```diff
@@ -155,8 +156,8 @@
             ).onErrorResumeNext(error -> {
                 Throwable recordTaskError = (Throwable) error; // Type inference issue
                 if (JobManagerException.hasErrorCode(recordTaskError, JobManagerException.ErrorCode.UnexpectedTaskState)) {
-                    // TODO More checking here. If it is actually running, we should kill it
-                    logger.info("Not launching task, as it is no longer in Accepted state (probably killed): {}", v3Task.getId());
+                    logger.info("Not launching task: {} and removing from fenzo as it is no longer in Accepted state (probably killed)", v3Task.getId());
+                    removeTask(assignmentResult, fenzoTask);
```

Comment:
I understand it's not necessarily part of your changes, but I just noticed we may have the same problem in all cases below where we call `killBrokenV3Task()`. Should we remove the task from Fenzo in all cases we do `killBrokenV3Task()` as well?

Reply:
I guess it will be eventually removed from Fenzo by line 130 (`removing after not finding jobMgr for task`) sometime in the future when the task gets archived, or by the next scheduling iteration when we try to place a Task that is in `KillInitiated|Finished` states, but it would be probably better to not leak resources for an extra scheduling iteration and waste some cycles trying to place them

Reply:
I agree that this is problematic, but I am going to leave it the same for now as it will eventually get cleaned up. We will need to revisit this in the future.